src/lib/y2partitioner/dialogs/format_and_mount.rb

```diff
@@ -7,28 +7,42 @@ module Dialogs
     # Part of {Sequences::AddPartition} and {Sequences::EditBlkDevice}.
     # Formerly MiniWorkflowStepFormatMount
     class FormatAndMount < CWM::Dialog
-      # @param partition [Y2Storage::Partition] FIXME: unsure which type we want
-      def initialize(partition)
-        @partition = partition
+      # @param options [Y2Partitioner::FormatMountOptions]
+      def initialize(options)
         textdomain "storage"
 
-        @format_widget = Widgets::FormatOptions.new(@partition)
-        @mount_widget  = Widgets::MountOptions.new(@partition)
+        @options = options
       end
 
       def title
-        "Edit Partition #{@partition.name}"
+        "Edit Partition #{@options.name}"
       end
 
       def contents
         HVSquash(
           HBox(
-            @format_widget,
-            HSpacing(4),
-            @mount_widget
+            Widgets::FormatOptions.new(@options),
+            HSpacing(5),
+            Widgets::MountOptions.new(@options)
           )
         )
       end
+
+      def cwm_show
+        ret = nil
+
+        loop do
+          ret = super
+
+          break if ret != :redraw
+        end
+
+        ret
+      end
+
+      def skip_store_for
+        [:redraw]
```

Comment:
Currently I'm skipping the store, but we probably will need to store the values, just have to be sure there is no inconsistencies with the selected filesystem and the selected partition id with the rest of widgets.